File: 2023/domaci/41-textovy-dekomprimator/volfmatous/main.py
if __name__ == "__main__":
    text = input()
    compressing = input() == "compress"
    
    if compressing:
        current_letter_count = 1
    
        for i in range(len(text)):
            if i < len(text) - 1 and text[i] == text[i + 1]:
                current_letter_count += 1
            else:
                print(text[i], end="")
                
                if current_letter_count > 1:
                    print(current_letter_count, end="")
                
                current_letter_count = 1
    else:
        i = 0
        while i < len(text):
            print(text[i], end="")
    
            count_buffer = ""
            j = i
            while j < len(text):
                if j < len(text) - 1 and text[j + 1].isdigit():
                    count_buffer += text[j + 1]
                else:
                    if count_buffer:
                        count = int(count_buffer)
                        print(text[i] * (count - 1), end="")
    
                    i = j
                    break
                j += 1
    
            i += 1
    
    print()

File: 2023/domaci/41-textovy-dekomprimator/volfmatous/main.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String text = scanner.nextLine();
        boolean compressing = scanner.nextLine().equals("compress");

        if (compressing) {
            int currentLetterCount = 1;

            for (int i = 0; i < text.length(); i++) {
                if (i < text.length() - 1 && text.charAt(i) == text.charAt(i + 1)) {
                    currentLetterCount++;
                } else {
                    System.out.print(text.charAt(i));
                    if (currentLetterCount > 1) {
                        System.out.print(currentLetterCount);
                    }

                    currentLetterCount = 1;
                }
            }
        } else {
            for (int i = 0; i < text.length(); i++) {
                System.out.print(text.charAt(i));

                String countBuffer = "";
                for (int j = i; j < text.length(); j++) {
                    if (j < text.length() - 1 && Character.isDigit(text.charAt(j + 1))) {
                        countBuffer += text.charAt(j + 1);
                    } else {
                        if (!countBuffer.isEmpty()) {
                            int count = Integer.parseInt(countBuffer);
                            for (int k = 0; k < count - 1; k++) {
                                System.out.print(text.charAt(i));
                            }
                        }

                        i = j;
                        break;
                    }
                }
            }
        }

        System.out.println();
    }
}
